tests/php/test-class-amp-youtube-embed-handler.php

```diff
@@ -188,6 +190,57 @@ public function test__conversion( $source, $expected, $fallback_for_expected = n
 		}
 	}
 
+	/**
+	 * Gets the test data for test_get_video_id_from_url().
+	 *
+	 * @return array The test data.
+	 */
+	public function get_url_id_data() {
+		return [
+			'basic_url'                        => [
+				'https://www.youtube.com/watch?v=XOY3ZUO6P0k',
+				'XOY3ZUO6P0k',
+			],
+			'short_url'                        => [
+				'https://youtu.be/XOY3ZUO6P0k',
+				'XOY3ZUO6P0k',
+			],
+			'url_with_underscore'              => [
+				'https://www.youtube.com/watch?v=CMrv_D78oxY',
+				'CMrv_D78oxY',
+			],
+			'short_url_with_underscore'        => [
+				'https://youtu.be/CMrv_D78oxY',
+				'CMrv_D78oxY',
+			],
+			'url_with_hyphen'                  => [
+				'https://www.youtube.com/watch?v=xo68-iWaKv8' . PHP_EOL,
+				'xo68-iWaKv8',
+			],
+			'url_with_hyphen_and_query_string' => [
+				'https://www.youtube.com/watch?v=xo68-iWaKv8&w=800&h=400' . PHP_EOL,
+				'xo68-iWaKv8',
+			],
+		];
+	}
+
+	/**
+	 * Tests get_video_id_from_url.
+	 *
+	 * @dataProvider get_url_id_data
+	 * @covers AMP_YouTube_Embed_Handler::get_video_id_from_url()
+	 *
+	 * @param string $url The URL to test.
+	 * @param string $expected The expected result.
+	 * @throws ReflectionException If a reflection of the object is not possible.
+	 */
+	public function test_get_video_id_from_url( $url, $expected ) {
+		$this->assertEquals(
+			$expected,
+			$this->call_private_method( $this->handler, 'get_video_id_from_url', [ $url ] )
```

Comment:
Testing this private method isn't ideal. It'd be better to add test cases to `test__conversion()`

But that would also require mocking 2 more [HTTP responses](https://github.com/ampproject/amp-wp/pull/4508/files#diff-23071a193ac2ab7d9be3c9eb4bdebac6R30), and might be overkill.

Reply:
Ah, good idea